src/main/java/cloud/eppo/ConfigurationStore.java

```diff
@@ -1,53 +1,106 @@
 package cloud.eppo;
 
-import cloud.eppo.ufc.dto.BanditParameters;
-import cloud.eppo.ufc.dto.FlagConfig;
-import cloud.eppo.ufc.dto.FlagConfigResponse;
+import cloud.eppo.ufc.dto.*;
 import cloud.eppo.ufc.dto.adapters.EppoModule;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Map;
+import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class ConfigurationStore {
   private static final Logger log = LoggerFactory.getLogger(ConfigurationStore.class);
   private final ObjectMapper mapper = new ObjectMapper().registerModule(EppoModule.eppoModule());
 
-  private ConcurrentHashMap<String, FlagConfig> flags;
-  private ConcurrentHashMap<String, BanditParameters> banditParameters; // TODO: bandit stuff
-  // TODO: another map of bandit flag variations to handle no action case
-  static ConfigurationStore instance = null;
+  private Map<String, FlagConfig> flags;
+  private Map<String, BanditReference> banditReferences;
+  private Map<String, BanditParameters> banditParameters;
 
   public ConfigurationStore() {
-    // TODO: handle caching
+    flags = new ConcurrentHashMap<>();
+    banditReferences = new ConcurrentHashMap<>();
+    banditParameters = new ConcurrentHashMap<>();
   }
 
-  public void setFlagsFromJsonString(String jsonString) throws JsonProcessingException {
-    FlagConfigResponse config = mapper.readValue(jsonString, FlagConfigResponse.class);
+  public void setFlagsFromJsonString(String jsonString) {
+    FlagConfigResponse config;
+
+    try {
+      config = mapper.readValue(jsonString, FlagConfigResponse.class);
+    } catch (JsonProcessingException e) {
+      log.error("Unable to parse flag configuration response");
+      throw new RuntimeException(e);
+    }
+
     if (config == null || config.getFlags() == null) {
       log.warn("Flags missing in configuration response");
       flags = new ConcurrentHashMap<>();
+      banditReferences = new ConcurrentHashMap<>();
     } else {
       // TODO: atomic flags to prevent clobbering like android does
       // Record that flags were set from a response so we don't later clobber them with a
       // slow cache read
       flags = new ConcurrentHashMap<>(config.getFlags());
-      log.debug("Loaded " + flags.size() + " flags from configuration response");
+      banditReferences = new ConcurrentHashMap<>(config.getBanditReferences());
```

Comment:
`banditReferences` is a top-level UFC property containing some information on any bandits at play (but not their model data, which is provided separately)